### PR TITLE
Snow 1304163 notebook create command

### PR DESCRIPTION
**Pull request by sfc-gh-mraba**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [x] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [x] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
Added support for notebook creation from stage `snow notebook create`

### CODE REVIEW

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 ## Deprecations
 
 ## New additions
+ * Add `snow notebook create` command
```
Comment (sfc-gh-turbaszek) [inline RELEASE-NOTES.md:7]: Please remove it from release notes, notebooks are not yet in public preview.

Comment (sfc-gh-mraba) [reply RELEASE-NOTES.md]: ✅

File: src/snowflake/cli/plugins/notebook/commands.py [hunk 2]
```diff
@@ -15,6 +18,11 @@
 log = logging.getLogger(__name__)
 
 NOTEBOOK_IDENTIFIER = identifier_argument(sf_object="notebook", example="MY_NOTEBOOK")
+NotebookFile: NotebookStagePath = typer.Option(
+    "--notebook-file",
+    "-f",
+    help="Stage path to notebook file.",
```
Comment (sfc-gh-turbaszek) [inline src/snowflake/cli/plugins/notebook/commands.py:24]: Please add example to help

File: src/snowflake/cli/plugins/notebook/commands.py [hunk 3]
```diff
@@ -49,3 +57,16 @@ def open_cmd(
...
+def create(
+    identifier: Annotated[NotebookName, NOTEBOOK_IDENTIFIER],
+    notebook_file: Annotated[NotebookStagePath, NotebookFile],
+    **options,
+):
+    """Creates notebook from stage."""
+    results = NotebookManager().create(
+        notebook_name=identifier, notebook_file=notebook_file
+    )
+    return MultipleResults(QueryResult(c) for c in results)
```
Comment (sfc-gh-turbaszek) [inline src/snowflake/cli/plugins/notebook/commands.py:72]: Can we try to return link to the notebook? Would it make sense?

File: src/snowflake/cli/plugins/notebook/manager.py [hunk 4]
```diff
@@ -1,16 +1,48 @@
+from pathlib import Path
+
+from snowflake.cli.api.cli_global_context import cli_context
 from snowflake.cli.api.identifiers import FQN
 from snowflake.cli.api.sql_execution import SqlExecutionMixin
 from snowflake.cli.plugins.connection.util import make_snowsight_url
 
+from .exceptions import NotebookStagePathError
+from .types import NotebookName, NotebookStagePath
```
Comment (sfc-gh-turbaszek) [inline src/snowflake/cli/plugins/notebook/manager.py:9]: Please use absolute imports.

File: src/snowflake/cli/plugins/notebook/manager.py [hunk 5]
```diff
@@ -1,16 +1,48 @@
...
+
+    def create(self, notebook_name: NotebookName, notebook_file: NotebookStagePath):
+        stage_path = self._parse_stage_path(notebook_file)
+
+        create_query = (
+            f"CREATE OR REPLACE NOTEBOOK {notebook_name.upper()}"
+            f" FROM '{stage_path.parent}'"
+            f" QUERY_WAREHOUSE = '{cli_context.connection.warehouse}'"
+            f" MAIN_FILE = '{stage_path.name}';"
+        )
```
Comment (sfc-gh-turbaszek) [inline src/snowflake/cli/plugins/notebook/manager.py:42]: Can you use multiline string? I think this is what we do in many places. SQL execution helpers will dedent the text.

File: src/snowflake/cli/plugins/notebook/manager.py [hunk 6]
```diff
@@ -1,16 +1,48 @@
...
+    def _parse_stage_path(notebook_file: NotebookName) -> Path:
+        if not notebook_file.endswith(".ipynb"):
+            raise NotebookStagePathError(notebook_file)
+        stage_path = Path(notebook_file)
+        if len(stage_path.parts) < 2:
+            raise NotebookStagePathError(notebook_file)
+
+        return stage_path
+
+    def create(self, notebook_name: NotebookName, notebook_file: NotebookStagePath):
```
Comment (sfc-gh-turbaszek) [inline src/snowflake/cli/plugins/notebook/manager.py:34]: Please change `notebook_name` to FQN. Current implementation will cause problems with quoted identifiers.

File: src/snowflake/cli/plugins/notebook/manager.py (hunk 4 above)
Comment (sfc-gh-mraba) [reply src/snowflake/cli/plugins/notebook/manager.py]: 🫡

File: src/snowflake/cli/plugins/notebook/manager.py (hunk 5 above)
Comment (sfc-gh-mraba) [reply src/snowflake/cli/plugins/notebook/manager.py]: 🫡

File: src/snowflake/cli/plugins/notebook/commands.py (hunk 2 above)
Comment (sfc-gh-mraba) [reply src/snowflake/cli/plugins/notebook/commands.py]: 🫡

File: src/snowflake/cli/plugins/notebook/manager.py (hunk 6 above)
Comment (sfc-gh-mraba) [reply src/snowflake/cli/plugins/notebook/manager.py:35]: 🫡

File: src/snowflake/cli/plugins/notebook/commands.py (hunk 3 above)
Comment (sfc-gh-mraba) [reply src/snowflake/cli/plugins/notebook/commands.py:72]: Sure.
Should we add link as next position to result or replace result with link?

Comment (sfc-gh-turbaszek) [reply src/snowflake/cli/plugins/notebook/commands.py:72]: The results is meaningless for now, so we should do similar think as we do in streamlit deploy

Comment (sfc-gh-mraba) [reply src/snowflake/cli/plugins/notebook/commands.py]: 🫡

**Pull request merged**